src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Managed.cs

```diff
@@ -218,6 +238,72 @@ public async Task ConnectAsync(Uri uri, CancellationToken cancellationToken, Cli
             }
         }
 
+        private static WebSocketDeflateOptions ParseDeflateOptions(ReadOnlySpan<char> extension, WebSocketDeflateOptions original)
+        {
+            var options = new WebSocketDeflateOptions();
+
+            while (true)
+            {
+                int end = extension.IndexOf(';');
+                ReadOnlySpan<char> value = (end >= 0 ? extension[..end] : extension).Trim();
+
+                if (!value.IsEmpty)
```

Comment:
We prefer to use Length to differentiate between empty spans and null pointers.

```suggestion
                if (value.Length == 0)
```